File: clients/python/src/live_mutex_client/smoke.py
"""
Smoke test for the Python live-mutex client.

Run with `python -m live_mutex_client.smoke` after the broker is
listening on `127.0.0.1:6970` (override with `LMX_HOST`/`LMX_PORT`).
Exits 0 on success, 1 on assertion failure.
"""

from __future__ import annotations

import asyncio
import os
import sys

from . import Client


async def main() -> int:
    host = os.environ.get("LMX_HOST", "127.0.0.1")
    port = int(os.environ.get("LMX_PORT", "6970"))

    async with await Client.connect(host, port) as client:
        g1 = await client.acquire("py-smoke", ttl_ms=5_000)
        print(f"acquire #1: lock_uuid={g1.lock_uuid} fencing_token={g1.fencing_token}")
        assert g1.fencing_token and g1.fencing_token >= 1, "missing fencing token"
        await client.release("py-smoke", g1.lock_uuid)

        g2 = await client.acquire("py-smoke")
        print(f"acquire #2: lock_uuid={g2.lock_uuid} fencing_token={g2.fencing_token}")
        assert g2.fencing_token and g1.fencing_token and g2.fencing_token > g1.fencing_token, \
            "fencing tokens must be strictly monotonic per key"
        await client.release("py-smoke", g2.lock_uuid)

        many = await client.acquire_many(["py-many-a", "py-many-b", "py-many-c"], ttl_ms=5_000)
        print(f"acquire_many: lock_uuid={many.lock_uuid} fencing_tokens={many.fencing_tokens}")
        assert len(many.fencing_tokens) == 3, "expected one token per key"
        await client.release_many(many.lock_uuid)

    print("\u2705 python client smoke test passed")
    return 0


if __name__ == "__main__":
    sys.exit(asyncio.run(main()))

File: clients/dart/example/smoke.dart
import 'dart:io';

import 'package:live_mutex_client/live_mutex_client.dart';

Future<void> main() async {
  final host = Platform.environment['LMX_HOST'] ?? '127.0.0.1';
  final port = int.tryParse(Platform.environment['LMX_PORT'] ?? '') ?? 6970;

  final client = await Client.connect(host: host, port: port);

  try {
    final g1 = await client.acquire('dart-smoke', ttlMs: 5000);
    print('acquire #1: lockUuid=${g1.lockUuid} fencingToken=${g1.fencingToken}');
    if (g1.fencingToken == null || g1.fencingToken! < 1) {
      throw StateError('missing fencing token');
    }
    await client.release('dart-smoke', g1.lockUuid);

    final g2 = await client.acquire('dart-smoke');
    print('acquire #2: lockUuid=${g2.lockUuid} fencingToken=${g2.fencingToken}');
    if (g2.fencingToken == null || g2.fencingToken! <= g1.fencingToken!) {
      throw StateError('fencing tokens must be strictly monotonic per key');
    }
    await client.release('dart-smoke', g2.lockUuid);

    final many = await client.acquireMany(
      ['dart-many-a', 'dart-many-b', 'dart-many-c'],
      ttlMs: 5000,
    );
    print('acquire_many: lockUuid=${many.lockUuid} fencingTokens=${many.fencingTokens}');
    if (many.fencingTokens.length != 3) {
      throw StateError('expected one token per key');
    }
    await client.releaseMany(many.lockUuid);

    print('\u2705 dart client smoke test passed');
  } finally {
    await client.close();
  }
}
